### PR TITLE
implement overrides retriever (closes #10)

### LINKED ISSUE #10: Daemon should extract overrides from PR body

### PR DESCRIPTION
**Pull request by g-arjones**
Works pretty nicely..

I wanted to implement just using regex with lookbehind to get the task items but didn't manage so I loop thorough all lines past the "Depends on" anchor until I find a line that does not look like a task item. It also allows to have nested lists and mixing things that are not pull requests in the task list.

### CODE REVIEW

**Review by doudou — COMMENTED** [review]
Looks fine. One simplification, though, would be to use GitHub's PR branches instead of the "real" branch (i.e. refs/pull/{ID}/head). This ref stays there after the pull request is merged, so it would allow you to even ignore whether tasks have been ticked or not (and continue working if the list is not up-to-date, since it usually is not)

**Comment by g-arjones** [comment]
Not necessary... client#pull_requests does not return closed pull requests... So I won't be getting closed pull requests anyway, even if the checkbox is not ticked

**Comment by g-arjones** [comment]
Please, approve if you are ok with this

**Review by doudou — APPROVED** [review]

**Pull request merged**
